mycore-ocfl/src/main/java/org/mycore/ocfl/metadata/MCROCFLXMLMetadataManager.java

```diff
@@ -317,7 +331,99 @@ public IntStream getStoredIDs(String project, String type) throws MCRPersistence
 
     @Override
     public int getHighestStoredID(String project, String type) {
-        return getStoredIDs(project, type).max().orElse(0);
+        int highestStoredID = 0;
+        int maxDepth = Integer.MAX_VALUE;
+        MCROCFLRepositoryProvider oclfRepoProvider
+            = MCRConfiguration2.getSingleInstanceOf("MCR.OCFL.Repository." + repositoryKey)
+                .map(MCROCFLRepositoryProvider.class::cast).orElseThrow();
+
+        OcflExtensionConfig config = oclfRepoProvider.getExtensionConfig();
+        Path basePath = null;
+
+        // optimization for known layouts
+        if (Objects.equals(config.getExtensionName(), MCRStorageLayoutExtension.EXTENSION_NAME)) {
+            maxDepth = ((MCRStorageLayoutConfig) config).getSlotLayout().split("-").length;
+            basePath = ((MCROCFLMCRRepositoryProvider) oclfRepoProvider).getRepositoryRoot()
+                .resolve(MCROCFLObjectIDPrefixHelper.MCROBJECT.replace(":", ""))
+                .resolve(project).resolve(type);
+            highestStoredID = traverseMCRStorageDirectory(basePath, maxDepth);
+        } else if (Objects.equals(config.getExtensionName(),
+            HashedNTupleIdEncapsulationLayoutExtension.EXTENSION_NAME)) {
+            maxDepth = ((HashedNTupleIdEncapsulationLayoutConfig) config).getNumberOfTuples() + 1;
+            basePath = ((MCROCFLHashRepositoryProvider) oclfRepoProvider).getRepositoryRoot();
+        } else {
+            //for other repository provider implementation start with root directory
+            basePath = MCRConfiguration2.getString("MCR.OCFL.Repository." + repositoryKey + ".RepositoryRoot")
+                .map(Path::of).orElse(null);
+        }
+
+        if (basePath != null && highestStoredID == 0) {
```

Comment:
After talking with robert about it, we found it, that it _might_ be possible that `traverseMCRStorageDirectory` _could_ return `-1`, so in that case this would then also run into the fallback:
```suggestion
        if (basePath != null && highestStoredID <= 0) {
```